packages/bits/src/lib/wizard-v2/wizard-step-header/wizard-step-header.component.ts

```diff
@@ -65,54 +101,62 @@ export class WizardStepHeaderComponent extends CdkStepHeader implements AfterVie
     }
 
     /** Focuses the step header. */
-    focus(): void {
+    public focus(): void {
         this._focusMonitor.focusVia(this._elementRef, "program");
     }
 
     /** Returns string label of given step if it is a text label. */
-    _stringLabel(): string | null {
+    public get stringLabel(): string | null {
         return this.label instanceof WizardStepLabelDirective ? null : this.label;
     }
 
     /** Returns WizardStepLabel if the label of given step is a template label. */
-    _templateLabel(): WizardStepLabelDirective | null {
+    public get templateLabel(): WizardStepLabelDirective | null {
         return this.label instanceof WizardStepLabelDirective ? this.label : null;
     }
 
-    _getDefaultTextForState(state: StepState): string {
-        let returnedState: StepState = state;
+    public get currentStepStateIconConfig(): IWizardCurrentStepStatusIconConfig | undefined {
 
-        switch (state) {
-            case "number":
-                returnedState = `${this.index + 1}`;
-                break;
-
-            case "edit":
-                returnedState = "create";
-                break;
-
-            case "error":
-                returnedState = "warning";
-                break;
-
-            default:
+        if (this.stepState === "number") {
+            return {
+                icon: this.wizardConfig.stepStateIcons?.icons?.initial,
+                color: this.wizardConfig.stepStateIcons?.colors?.initial,
+            };
+        }
+        
+        if (this.stepState === "done") {
+            return {
+                icon: this.wizardConfig.stepStateIcons?.icons?.visited,
+                color: this.wizardConfig.stepStateIcons?.colors?.visited,
+            };
+        }
+        
+        if (this.stepState === "edit") {
+            return {
+                icon: this.wizardConfig.stepStateIcons?.icons?.active,
+                color: this.wizardConfig.stepStateIcons?.colors?.active,
+            };
+        }
+        
+        if (this.stepState === "error") {
```

Comment:
nit: mb better use here STEP_STATE instead string?

Reply:
that's a good point